src/Microsoft.Azure.SignalR.Management/HubInstanceFactories/MultiEndpointConnectionContainerFactory.cs

```diff
@@ -27,31 +24,16 @@ public MultiEndpointConnectionContainerFactory(IServiceConnectionFactory connect
             _router = router;
         }
 
-        public MultiEndpointServiceConnectionContainer GetOrCreate(string hubName, ILoggerFactory loggerFactoryPerHub = null)
+        public MultiEndpointServiceConnectionContainer CreateAndStart(string hubName)
         {
-            return _hubConnectionContainers.GetOrAdd(hubName, (hubName) => Create(hubName, loggerFactoryPerHub)).Value;
-        }
-
-        /// <summary>
-        /// creates a lazy container object and starts it after creation.
-        /// </summary>
-        /// <remarks>
-        /// actually the <see cref="MultiEndpointServiceConnectionContainer"/> itself has mechanism to ensure being started only once, using Lazy and starting the container here just to avoid starting it everywhere else
-        /// </remarks>
-        private Lazy<MultiEndpointServiceConnectionContainer> Create(string hubName, ILoggerFactory loggerFactoryPerHub = null)
-        {
-            var loggerFactory = loggerFactoryPerHub ?? _loggerFactory;
-            return new Lazy<MultiEndpointServiceConnectionContainer>(() =>
-            {
-                var container = new MultiEndpointServiceConnectionContainer(
+            var container = new MultiEndpointServiceConnectionContainer(
                 hubName,
-                endpoint => new WeakServiceConnectionContainer(_connectionFactory, _connectionCount, endpoint, loggerFactory.CreateLogger<WeakServiceConnectionContainer>()),
+                endpoint => new WeakServiceConnectionContainer(_connectionFactory, _connectionCount, endpoint, _loggerFactory.CreateLogger<WeakServiceConnectionContainer>()),
                 _endpointManager,
                 _router,
-                loggerFactory);
-                container.StartAsync();
-                return container;
-            }, true);
+                _loggerFactory);
+            container.StartAsync();
```

Comment:
so we don't know when the container is started or failed to start when cx is already sending messages?

Reply:
We wait [here](https://github.com/Azure/azure-signalr/pull/1184/files#diff-8c62721fa2e48aa7b5a9d16291699f40f45fa98b2e436eaf3e757a2e2356e45eR50)

Reply:
move the startAsync to the line before await can be clearer?
rename the method to Create()?